src/caja-desktop-window.c

```diff
@@ -191,11 +225,47 @@ caja_desktop_window_new (CajaApplication *application,
     /* will cause the desktop window to open at the wrong size in gtk 3.20 */
     gtk_window_set_default_size (GTK_WINDOW (window), -1, -1);
 
+    /*For wayland only
+     *Code taken from gtk-layer-shell simple-example.c
+     */
+#ifdef HAVE_WAYLAND
+    if (GDK_IS_WAYLAND_DISPLAY (display))
+    {
+        GtkWindow *gtkwin;
+        gtkwin = (GTK_WINDOW(window));
+
+        /* Before the window is first realized, set it up to be a layer surface */
+        gtk_layer_init_for_window (gtkwin);
+
+        /* Order below normal windows */
+        gtk_layer_set_layer (gtkwin, GTK_LAYER_SHELL_LAYER_BOTTOM);
+
+        gtk_layer_set_namespace (gtkwin, "desktop");
+
+        /*Anchor the desktop to all four corners
+         *This is much simpler than on x11 and
+         *should always render the desktop across
+         *all of the screen
+         */
+        gtk_layer_set_anchor (gtkwin,GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+        gtk_layer_set_anchor (gtkwin,GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
+        gtk_layer_set_anchor (gtkwin,GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+        gtk_layer_set_anchor (gtkwin,GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
+
+        /*Enable keyboard use on the desktop*/
+        gtk_layer_set_keyboard_mode (gtkwin, GTK_LAYER_SHELL_KEYBOARD_MODE_ON_DEMAND);
+    }
+#endif
     /* Special sawmill setting*/
     GdkWindow *gdkwin;
-    gtk_widget_realize (GTK_WIDGET (window));
+    if  ((GDK_IS_X11_DISPLAY (display)))
```

Comment:
only one space after `if`, and no need to double parentheses around the condition
```suggestion
    if (GDK_IS_X11_DISPLAY (display))
```

Reply:
found and fixed